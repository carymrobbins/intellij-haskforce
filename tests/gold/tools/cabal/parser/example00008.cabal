Name:             ShellCheck
Version:          0.4.3
Synopsis:         Shell script analysis tool
License:          GPL-3
License-file:     LICENSE
Category:         Static Analysis
Author:           Vidar Holen
Maintainer:       vidar@vidarholen.net
Homepage:         http://www.shellcheck.net/
Build-Type:       Custom
Cabal-Version:    >= 1.8
Bug-reports:      https://github.com/koalaman/shellcheck/issues
Description:
  The goals of ShellCheck are:
  .
  * To point out and clarify typical beginner's syntax issues,
    that causes a shell to give cryptic error messages.
  .
  * To point out and clarify typical intermediate level semantic problems,
    that causes a shell to behave strangely and counter-intuitively.
  .
  * To point out subtle caveats, corner cases and pitfalls, that may cause an
    advanced user's otherwise working script to fail under future circumstances.

Extra-Source-Files:
    -- documentation
    README.md
    shellcheck.1.md
    -- built with a cabal sdist hook
    shellcheck.1
    -- tests
    test/shellcheck.hs

source-repository head
    type: git
    location: git://github.com/koalaman/shellcheck.git

library
    build-depends:
      base >= 4 && < 5,
      containers,
      directory,
      json,
      mtl >= 2.2.1,
      parsec,
      regex-tdfa,
      QuickCheck >= 2.7.4,
      -- When cabal supports it, move this to setup-depends:
      process
    exposed-modules:
      ShellCheck.AST
      ShellCheck.ASTLib
      ShellCheck.Analytics
      ShellCheck.Analyzer
      ShellCheck.AnalyzerLib
      ShellCheck.Checker
      ShellCheck.Checks.Commands
      ShellCheck.Data
      ShellCheck.Formatter.Format
      ShellCheck.Formatter.CheckStyle
      ShellCheck.Formatter.GCC
      ShellCheck.Formatter.JSON
      ShellCheck.Formatter.TTY
      ShellCheck.Interface
      ShellCheck.Parser
      ShellCheck.Regex
    other-modules:
      Paths_ShellCheck

executable shellcheck
    build-depends:
      ShellCheck,
      base >= 4 && < 5,
      containers,
      directory,
      json,
      mtl >= 2.2.1,
      parsec,
      regex-tdfa,
      QuickCheck >= 2.7.4
    main-is: shellcheck.hs

test-suite test-shellcheck
    type: exitcode-stdio-1.0
    build-depends:
      ShellCheck,
      base >= 4 && < 5,
      containers,
      directory,
      json,
      mtl >= 2.2.1,
      parsec,
      regex-tdfa,
      QuickCheck >= 2.7.4
    main-is: test/shellcheck.hs
