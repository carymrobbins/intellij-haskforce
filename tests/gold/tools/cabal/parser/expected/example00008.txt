Cabal File
  PKG_NAME
    PsiElement(NAME_KEY)('Name')
    PsiElement(COLON)(':')
    FREEFORM
      PsiElement(FREEFORM_LINE)('ShellCheck')
  PKG_VERSION
    PsiElement(VERSION_KEY)('Version')
    PsiElement(COLON)(':')
    FREEFORM
      PsiElement(FREEFORM_LINE)('0.4.3')
  SYNOPSIS
    PsiElement(SYNOPSIS_KEY)('Synopsis')
    PsiElement(COLON)(':')
    FREEFORM
      PsiElement(FREEFORM_LINE)('Shell script analysis tool')
  LICENSE
    PsiElement(LICENSE_KEY)('License')
    PsiElement(COLON)(':')
    FREEFORM
      PsiElement(FREEFORM_LINE)('GPL-3')
  LICENSE_FILE
    PsiElement(LICENSE_FILE_KEY)('License-file')
    PsiElement(COLON)(':')
    FREEFORM
      PsiElement(FREEFORM_LINE)('LICENSE')
  CATEGORY
    PsiElement(CATEGORY_KEY)('Category')
    PsiElement(COLON)(':')
    FREEFORM
      PsiElement(FREEFORM_LINE)('Static Analysis')
  AUTHOR
    PsiElement(AUTHOR_KEY)('Author')
    PsiElement(COLON)(':')
    FREEFORM
      PsiElement(FREEFORM_LINE)('Vidar Holen')
  MAINTAINER
    PsiElement(MAINTAINER_KEY)('Maintainer')
    PsiElement(COLON)(':')
    FREEFORM
      PsiElement(FREEFORM_LINE)('vidar@vidarholen.net')
  HOMEPAGE
    PsiElement(HOMEPAGE_KEY)('Homepage')
    PsiElement(COLON)(':')
    FREEFORM
      PsiElement(FREEFORM_LINE)('http://www.shellcheck.net/')
  BUILD_TYPE
    PsiElement(BUILD_TYPE_KEY)('Build-Type')
    PsiElement(COLON)(':')
    FREEFORM
      PsiElement(FREEFORM_LINE)('Custom')
  CABAL_VERSION
    PsiElement(CABAL_VERSION_KEY)('Cabal-Version')
    PsiElement(COLON)(':')
    FREEFORM
      PsiElement(FREEFORM_LINE)('>= 1.8')
  BUG_REPORTS
    PsiElement(BUG_REPORTS_KEY)('Bug-reports')
    PsiElement(COLON)(':')
    FREEFORM
      PsiElement(FREEFORM_LINE)('https://github.com/koalaman/shellcheck/issues')
  DESCRIPTION
    PsiElement(DESCRIPTION_KEY)('Description')
    PsiElement(COLON)(':')
    FREEFORM
      PsiElement(FREEFORM_LINE)('The goals of ShellCheck are:')
      PsiElement(FREEFORM_LINE)('.')
      PsiElement(FREEFORM_LINE)('* To point out and clarify typical beginner's syntax issues,')
      PsiElement(FREEFORM_LINE)('that causes a shell to give cryptic error messages.')
      PsiElement(FREEFORM_LINE)('.')
      PsiElement(FREEFORM_LINE)('* To point out and clarify typical intermediate level semantic problems,')
      PsiElement(FREEFORM_LINE)('that causes a shell to behave strangely and counter-intuitively.')
      PsiElement(FREEFORM_LINE)('.')
      PsiElement(FREEFORM_LINE)('* To point out subtle caveats, corner cases and pitfalls, that may cause an')
      PsiElement(FREEFORM_LINE)('advanced user's otherwise working script to fail under future circumstances.')
  EXTRA_SOURCE_FILES
    PsiElement(EXTRA_SOURCE_FILES_KEY)('Extra-Source-Files')
    PsiElement(COLON)(':')
    PsiComment(COMMENT)('    -- documentation')
    FREEFORM
      PsiElement(FREEFORM_LINE)('README.md')
      PsiElement(FREEFORM_LINE)('shellcheck.1.md')
      PsiComment(COMMENT)('    -- built with a cabal sdist hook')
      PsiElement(FREEFORM_LINE)('shellcheck.1')
      PsiComment(COMMENT)('    -- tests')
      PsiElement(FREEFORM_LINE)('test/shellcheck.hs')
  SOURCE_REPO
    PsiElement(SOURCE_REPO_KEY)('source-repository')
    PsiElement(SOURCE_REPO_NAME)('head')
    SOURCE_REPO_TYPE
      PsiElement(TYPE_KEY)('type')
      PsiElement(COLON)(':')
      FREEFORM
        PsiElement(FREEFORM_LINE)('git')
    SOURCE_REPO_LOCATION
      PsiElement(LOCATION_KEY)('location')
      PsiElement(COLON)(':')
      FREEFORM
        PsiElement(FREEFORM_LINE)('git://github.com/koalaman/shellcheck.git')
  LIBRARY
    PsiElement(LIBRARY_KEY)('library')
    BUILD_DEPENDS_KEY
      PsiElement(BUILD_DEPENDS_KEY)('build-depends')
      PsiElement(COLON)(':')
      DEPENDENCIES
        DEPENDENCY
          PsiElement(DEPENDENCY_NAME)('base')
          DEPENDENCY_VERSION
            PsiElement(>=)('>=')
            PsiElement(NUMBERS)('4')
            PsiElement(&&)('&&')
            PsiElement(<)('<')
            PsiElement(NUMBERS)('5')
        PsiElement(,)(',')
        DEPENDENCY
          PsiElement(DEPENDENCY_NAME)('containers')
        PsiElement(,)(',')
        DEPENDENCY
          PsiElement(DEPENDENCY_NAME)('directory')
        PsiElement(,)(',')
        DEPENDENCY
          PsiElement(DEPENDENCY_NAME)('json')
        PsiElement(,)(',')
        DEPENDENCY
          PsiElement(DEPENDENCY_NAME)('mtl')
          DEPENDENCY_VERSION
            PsiElement(>=)('>=')
            PsiElement(NUMBERS)('2.2.1')
        PsiElement(,)(',')
        DEPENDENCY
          PsiElement(DEPENDENCY_NAME)('parsec')
        PsiElement(,)(',')
        DEPENDENCY
          PsiElement(DEPENDENCY_NAME)('regex-tdfa')
        PsiElement(,)(',')
        DEPENDENCY
          PsiElement(DEPENDENCY_NAME)('QuickCheck')
          DEPENDENCY_VERSION
            PsiElement(>=)('>=')
            PsiElement(NUMBERS)('2.7.4')
        PsiElement(,)(',')
        PsiComment(COMMENT)('      -- When cabal supports it, move this to setup-depends:')
        DEPENDENCY
          PsiElement(DEPENDENCY_NAME)('process')
    EXPOSED_MODULES
      PsiElement(EXPOSED_MODULES_KEY)('exposed-modules')
      PsiElement(COLON)(':')
      MODULE_LIST
        MODULE
          MODULE_PART
            PsiElement(IDENT)('ShellCheck')
          PsiElement(.)('.')
          MODULE_PART
            PsiElement(IDENT)('AST')
        MODULE
          MODULE_PART
            PsiElement(IDENT)('ShellCheck')
          PsiElement(.)('.')
          MODULE_PART
            PsiElement(IDENT)('ASTLib')
        MODULE
          MODULE_PART
            PsiElement(IDENT)('ShellCheck')
          PsiElement(.)('.')
          MODULE_PART
            PsiElement(IDENT)('Analytics')
        MODULE
          MODULE_PART
            PsiElement(IDENT)('ShellCheck')
          PsiElement(.)('.')
          MODULE_PART
            PsiElement(IDENT)('Analyzer')
        MODULE
          MODULE_PART
            PsiElement(IDENT)('ShellCheck')
          PsiElement(.)('.')
          MODULE_PART
            PsiElement(IDENT)('AnalyzerLib')
        MODULE
          MODULE_PART
            PsiElement(IDENT)('ShellCheck')
          PsiElement(.)('.')
          MODULE_PART
            PsiElement(IDENT)('Checker')
        MODULE
          MODULE_PART
            PsiElement(IDENT)('ShellCheck')
          PsiElement(.)('.')
          MODULE_PART
            PsiElement(IDENT)('Checks')
          PsiElement(.)('.')
          MODULE_PART
            PsiElement(IDENT)('Commands')
        MODULE
          MODULE_PART
            PsiElement(IDENT)('ShellCheck')
          PsiElement(.)('.')
          MODULE_PART
            PsiElement(IDENT)('Data')
        MODULE
          MODULE_PART
            PsiElement(IDENT)('ShellCheck')
          PsiElement(.)('.')
          MODULE_PART
            PsiElement(IDENT)('Formatter')
          PsiElement(.)('.')
          MODULE_PART
            PsiElement(IDENT)('Format')
        MODULE
          MODULE_PART
            PsiElement(IDENT)('ShellCheck')
          PsiElement(.)('.')
          MODULE_PART
            PsiElement(IDENT)('Formatter')
          PsiElement(.)('.')
          MODULE_PART
            PsiElement(IDENT)('CheckStyle')
        MODULE
          MODULE_PART
            PsiElement(IDENT)('ShellCheck')
          PsiElement(.)('.')
          MODULE_PART
            PsiElement(IDENT)('Formatter')
          PsiElement(.)('.')
          MODULE_PART
            PsiElement(IDENT)('GCC')
        MODULE
          MODULE_PART
            PsiElement(IDENT)('ShellCheck')
          PsiElement(.)('.')
          MODULE_PART
            PsiElement(IDENT)('Formatter')
          PsiElement(.)('.')
          MODULE_PART
            PsiElement(IDENT)('JSON')
        MODULE
          MODULE_PART
            PsiElement(IDENT)('ShellCheck')
          PsiElement(.)('.')
          MODULE_PART
            PsiElement(IDENT)('Formatter')
          PsiElement(.)('.')
          MODULE_PART
            PsiElement(IDENT)('TTY')
        MODULE
          MODULE_PART
            PsiElement(IDENT)('ShellCheck')
          PsiElement(.)('.')
          MODULE_PART
            PsiElement(IDENT)('Interface')
        MODULE
          MODULE_PART
            PsiElement(IDENT)('ShellCheck')
          PsiElement(.)('.')
          MODULE_PART
            PsiElement(IDENT)('Parser')
        MODULE
          MODULE_PART
            PsiElement(IDENT)('ShellCheck')
          PsiElement(.)('.')
          MODULE_PART
            PsiElement(IDENT)('Regex')
    OTHER_MODULES
      PsiElement(OTHER_MODULES_KEY)('other-modules')
      PsiElement(COLON)(':')
      MODULE_LIST
        MODULE
          MODULE_PART
            PsiElement(IDENT)('Paths_ShellCheck')
  EXECUTABLE
    PsiElement(EXECUTABLE_KEY)('executable')
    PsiElement(EXECUTABLE_NAME)('shellcheck')
    BUILD_DEPENDS_KEY
      PsiElement(BUILD_DEPENDS_KEY)('build-depends')
      PsiElement(COLON)(':')
      DEPENDENCIES
        DEPENDENCY
          PsiElement(DEPENDENCY_NAME)('ShellCheck')
        PsiElement(,)(',')
        DEPENDENCY
          PsiElement(DEPENDENCY_NAME)('base')
          DEPENDENCY_VERSION
            PsiElement(>=)('>=')
            PsiElement(NUMBERS)('4')
            PsiElement(&&)('&&')
            PsiElement(<)('<')
            PsiElement(NUMBERS)('5')
        PsiElement(,)(',')
        DEPENDENCY
          PsiElement(DEPENDENCY_NAME)('containers')
        PsiElement(,)(',')
        DEPENDENCY
          PsiElement(DEPENDENCY_NAME)('directory')
        PsiElement(,)(',')
        DEPENDENCY
          PsiElement(DEPENDENCY_NAME)('json')
        PsiElement(,)(',')
        DEPENDENCY
          PsiElement(DEPENDENCY_NAME)('mtl')
          DEPENDENCY_VERSION
            PsiElement(>=)('>=')
            PsiElement(NUMBERS)('2.2.1')
        PsiElement(,)(',')
        DEPENDENCY
          PsiElement(DEPENDENCY_NAME)('parsec')
        PsiElement(,)(',')
        DEPENDENCY
          PsiElement(DEPENDENCY_NAME)('regex-tdfa')
        PsiElement(,)(',')
        DEPENDENCY
          PsiElement(DEPENDENCY_NAME)('QuickCheck')
          DEPENDENCY_VERSION
            PsiElement(>=)('>=')
            PsiElement(NUMBERS)('2.7.4')
    MAIN_IS
      PsiElement(MAIN_IS_KEY)('main-is')
      PsiElement(COLON)(':')
      FREEFORM
        PsiElement(FREEFORM_LINE)('shellcheck.hs')
  TEST_SUITE
    PsiElement(TEST_SUITE_KEY)('test-suite')
    PsiElement(TEST_SUITE_NAME)('test-shellcheck')
    TEST_SUITE_TYPE
      PsiElement(TYPE_KEY)('type')
      PsiElement(COLON)(':')
      FREEFORM
        PsiElement(FREEFORM_LINE)('exitcode-stdio-1.0')
    BUILD_DEPENDS_KEY
      PsiElement(BUILD_DEPENDS_KEY)('build-depends')
      PsiElement(COLON)(':')
      DEPENDENCIES
        DEPENDENCY
          PsiElement(DEPENDENCY_NAME)('ShellCheck')
        PsiElement(,)(',')
        DEPENDENCY
          PsiElement(DEPENDENCY_NAME)('base')
          DEPENDENCY_VERSION
            PsiElement(>=)('>=')
            PsiElement(NUMBERS)('4')
            PsiElement(&&)('&&')
            PsiElement(<)('<')
            PsiElement(NUMBERS)('5')
        PsiElement(,)(',')
        DEPENDENCY
          PsiElement(DEPENDENCY_NAME)('containers')
        PsiElement(,)(',')
        DEPENDENCY
          PsiElement(DEPENDENCY_NAME)('directory')
        PsiElement(,)(',')
        DEPENDENCY
          PsiElement(DEPENDENCY_NAME)('json')
        PsiElement(,)(',')
        DEPENDENCY
          PsiElement(DEPENDENCY_NAME)('mtl')
          DEPENDENCY_VERSION
            PsiElement(>=)('>=')
            PsiElement(NUMBERS)('2.2.1')
        PsiElement(,)(',')
        DEPENDENCY
          PsiElement(DEPENDENCY_NAME)('parsec')
        PsiElement(,)(',')
        DEPENDENCY
          PsiElement(DEPENDENCY_NAME)('regex-tdfa')
        PsiElement(,)(',')
        DEPENDENCY
          PsiElement(DEPENDENCY_NAME)('QuickCheck')
          DEPENDENCY_VERSION
            PsiElement(>=)('>=')
            PsiElement(NUMBERS)('2.7.4')
    MAIN_IS
      PsiElement(MAIN_IS_KEY)('main-is')
      PsiElement(COLON)(':')
      FREEFORM
        PsiElement(FREEFORM_LINE)('test/shellcheck.hs')